### PR TITLE
Issue 105: Configure and run `black` and other pre-commit hooks (clean branch)

### PR DESCRIPTION
**Pull request by lucasmbrown-usds**
The first and third commits are setting up the config so you probably want to look at them directly to understand the config. The second commit runs all the pre-commits everywhere, which generates a huge amount of diff.

I squashed all the commits on the branch both before and after running everywhere to make the diff easier to follow. 

These hooks fix:

Trailing whitespace at end of lines
Newlines at end of file
Removes unused python imports
Removes unused variables
Reorders python imports alphabetically and by format
Runs black using the same config from pyproject.toml

Instructions in the README are at: [data/data-pipeline/README.md ](https://github.com/usds/justice40-tool/blob/7d6ca49afc6fddb3187616640861a1fafbb0aae7/data/data-pipeline/README.md#configuring-pre-commit-hooks)

### CODE REVIEW

**Review by vim-usds — APPROVED** [review]
LGTM!

**Review by mattbowen-usds — COMMENTED** [review]
A couple questions

File: data/data-pipeline/.pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+exclude: ^client|\.csv
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.3.0
+  hooks:
+    - id: end-of-file-fixer
+    - id: trailing-whitespace
+    - id: requirements-txt-fixer
```
Comment (mattbowen-usds) [inline data/data-pipeline/.pre-commit-config.yaml:8]: Just remembered, we don't have a requirements.txt

**Comment by lucasmbrown-usds** [comment]
FYI, I generated a score on this branch and a score on `narwhal`, compared them using the notebook `compare_two_score_files_for_differences.ipynb`, and there are zero differences!

**Review by mattbowen-usds — APPROVED** [review]
My previous questions still stand but I wouldn't block merging on any of them.

File: data/data-pipeline/.pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        ]
+
+- repo: https://github.com/pycqa/isort
+  rev: 5.10.1
+  hooks:
+    - id: isort
+      name: isort (python)
+      args:
+        [
+        "--force-single-line-imports",
```
[the remark this answers is not in the document]
Comment (lucasmbrown-usds) [reply data/data-pipeline/.pre-commit-config.yaml:28]: It's all about merge conflicts!! https://github.com/asottile/reorder_python_imports#why-this-style

**Pull request merged**